functions/sudope.fish

```diff
@@ -19,32 +27,32 @@ function sudope -d "Quickly toggle sudo prefix"
 
   switch (count $command_parts)
     case 3
-      # No "sudo".
+      # No su command.
 
-      # Add "sudo" to the beginning of the command, after any leading whitespace (if present).
-      commandline -r (string join \n (string join '' $command_parts[2] 'sudo ' $command_parts[3]) $command_rest)
+      # Add the su command to the beginning of the command, after any leading whitespace (if present).
+      commandline -r (string join \n (string join '' $command_parts[2] "$sudope_command " $command_parts[3]) $command_rest)
 
-      # Push the cursor position ahead if necessary (+5 for 'sudo ').
+      # Push the cursor position ahead if necessary
       test $cursor_position -ge (string length -- "$command_parts[2]")
-        and set cursor_position (math $cursor_position+5)
+        and set cursor_position (math $cursor_position+$sudope_command_length)
 
       # Place the cursor where it was (or where it should be).
       commandline -C $cursor_position
 
     case 5
-      # "sudo" is present in the beginning of the command.
+      # The su comamnd is present in the beginning of the command.
 
-      # Remove "sudo", leave any leading whitespace (if present).
+      # Remove the su command, leave any leading whitespace (if present).
       commandline -r (string join \n (string join '' $command_parts[2 5]) $command_rest)
 
-      # Push the cursor position back if appropriate ('sudo' and whitespace).
+      # Push the cursor position back if appropriate (su command and whitespace).
       set -l lead_length (string length -- "$command_parts[2]")
-      set -l sudo_length (string length -- "$command_parts[3]")
-      if test $cursor_position -ge (math $lead_length+$sudo_length)
-        # The cursor was after "sudo".
-        set cursor_position (math $cursor_position-$sudo_length)
+      set sudope_command_length (string length -- "$command_parts[3]")
```

Comment:
`set -l` here?

Reply:
Again from `help set`:
> If a variable is not explicitly set to be either universal, global or
       local, but has been previously defined, the previous variable scope is
       used.

Reply:
right. i thought we _wanted_ this one to be block scoped? do we want it to persist after this case block?

Reply:
In the contest of your reply above (https://github.com/oh-my-fish/plugin-sudope/pull/15#discussion_r491706413), it makes no sense to make it persistent. I'll fix all these issues in my next commits. Thanks.

Reply:
Fixed in 18d9fd2.